iotfunctions/anomaly.py

```diff
@@ -0,0 +1,278 @@
+# *****************************************************************************
+# Â© Copyright IBM Corp. 2018.  All Rights Reserved.
+#
+# This program and the accompanying materials
+# are made available under the terms of the Apache V2.0
+# which accompanies this distribution, and is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# *****************************************************************************
+
+'''
+The Built In Functions module contains preinstalled functions
+'''
+
+import datetime as dt
+import time
+from collections import OrderedDict
+import numpy as np
+import scipy as sp
+
+# for Spectral Analysis
+from scipy import signal
+
+# for KMeans
+import skimage as ski
+from skimage import util as skiutil # for nifty windowing
+from pyod.models.cblof import CBLOF
+
+import re
+import pandas as pd
+import logging
+import warnings
+import json
+from sqlalchemy import Column, Integer, String, Float, DateTime, Boolean, func
+from .base import (BaseTransformer, BaseEvent, BaseSCDLookup, BaseMetadataProvider, BasePreload, BaseDatabaseLookup,
+                   BaseDataSource, BaseDBActivityMerge, BaseSimpleAggregator)
+
+from .ui import (UISingle, UIMultiItem, UIFunctionOutSingle, UISingleItem, UIFunctionOutMulti, UIMulti, UIExpression,
+                 UIText, UIStatusFlag, UIParameters)
+
+from .util import adjust_probabilities, reset_df_index
+
+logger = logging.getLogger(__name__)
+PACKAGE_URL = 'git+https://github.com/ibm-watson-iot/functions.git@'
+_IS_PREINSTALLED = True
+
+
+class SpectralAnomalyScore(BaseTransformer):
+    '''
+    Employs spectral analysis to extract features from the time series data and to compute zscore from it
+    '''
+    def __init__(self, input_item, windowsize, output_item):
```

Comment:
If the library allows do declare defaults we can do that

`def __init__(self, input_item, windowsize=12, output_item):`

Reply:
good idea - btw, I've modified the code to fix the kmeans approach.
you might notice that there is some code overlap, maybe we can simplify things..